prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -29,14 +40,41 @@
 from prediction_market_agent.utils import APIKeys
 
 
+def build_wxdai_functions(keys: APIKeys) -> list[Function]:
+    functions = []
+
+    wrapped_xdai = WrappedxDaiContract()
+    contract_address = Web3.to_checksum_address(wrapped_xdai.address)
+    contract_class_converter = ContractClassConverter(
+        contract_address=contract_address, contract_name=wrapped_xdai.__class__.__name__
+    )
+    function_types_to_classes = (
+        contract_class_converter.create_classes_from_smart_contract()
+    )
+
+    view_classes = function_types_to_classes[AbiItemStateMutabilityEnum.VIEW]
+    functions.extend([clz() for clz in view_classes])
+
+    payable_classes = function_types_to_classes[AbiItemStateMutabilityEnum.PAYABLE]
+    non_payable_classes = function_types_to_classes[
+        AbiItemStateMutabilityEnum.NON_PAYABLE
+    ]
+    for clz in payable_classes + non_payable_classes:
+        functions.append(clz(keys=keys))
+
+    return functions
+
+
 def build_agent_functions(
     agent: Agent,
     market_type: MarketType,
     keys: APIKeys,
     allow_stop: bool,
     long_term_memory: LongTermMemoryTableHandler | None,
     model: str,
+    build_dynamic_contract_functions: bool = False,
 ) -> list[Function]:
+    logger.error("entered build agent functions")
```

Comment:
Why is there a error log for `build_agent_functions` ?

Reply:
debugging session leftover. Removed.